### PR TITLE
Feature/rework ordering gate with shared model

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
### Description of the Change
Rework ordering gate with shared_model.

### Benefits
One step closer to full relocation on shared_model.

### Possible Drawbacks 
no

### Usage Examples or Tests *[optional]*
no

### Alternate Designs *[optional]*
no

### CODE REVIEW

File: test/module/irohad/ordering/ordering_gate_test.cpp [hunk 1]
```diff
@@ -172,8 +172,11 @@ TEST(OrderingGateQueueBehaviour, SendManyProposals) {
       make_test_subscriber<CallExact>(ordering_gate.on_proposal(), 2);
   wrapper_after.subscribe();
 
-  ordering_gate.onProposal(iroha::model::Proposal{{}});
-  ordering_gate.onProposal(iroha::model::Proposal{{}});
+  auto proposal = std::make_shared<shared_model::proto::Proposal>(
+      TestProposalBuilder().build());
+
+  ordering_gate.onProposal(proposal);
+  ordering_gate.onProposal(proposal);
```
Comment (victordrobny) [inline test/module/irohad/ordering/ordering_gate_test.cpp]: You pass shared pointer to the same proposal twice that will be moved somewhere

File: irohad/ordering/impl/ordering_gate_transport_grpc.cpp [hunk 2]
```diff
@@ -24,17 +28,21 @@ grpc::Status OrderingGateTransportGrpc::onProposal(
...
+  auto proposal = std::make_shared<shared_model::proto::Proposal>(
+      shared_model::proto::ProposalBuilder()
+                      .transactions(transactions)
+                      .height(request->height())
+                      .createdTime(request->created_time())
+                      .build());
+
   if (not subscriber_.expired()) {
-    subscriber_.lock()->onProposal(std::move(proposal));
+    subscriber_.lock()->onProposal(proposal);
```
Comment (victordrobny) [inline irohad/ordering/impl/ordering_gate_transport_grpc.cpp]: Why did you remove std::move?

File: test/module/irohad/simulator/simulator_test.cpp [hunk 3]
```diff
@@ -61,6 +63,8 @@ class SimulatorTest : public ::testing::Test {
   std::shared_ptr<MockOrderingGate> ordering_gate;
   std::shared_ptr<iroha::model::MockCryptoProvider> crypto_provider;
 
+  std::shared_ptr<shared_model::interface::Proposal> empty_proposal;
```
Comment (victordrobny) [inline test/module/irohad/simulator/simulator_test.cpp]: It seems to be unused variable

File: irohad/ordering/impl/ordering_gate_impl.hpp [hunk 4]
```diff
@@ -20,15 +20,19 @@
 
 #include "network/ordering_gate.hpp"
 
-#include <atomic>
-
 #include <tbb/concurrent_queue.h>
+#include <atomic>
```
Comment (nickaleks) [inline irohad/ordering/impl/ordering_gate_impl.hpp]: Standard includes should go before library ones

File: irohad/ordering/impl/ordering_gate_transport_grpc.cpp [hunk 5]
```diff
@@ -24,17 +28,21 @@ grpc::Status OrderingGateTransportGrpc::onProposal(
     ::google::protobuf::Empty *response) {
   log_->info("receive proposal");
 
-  auto transactions = decltype(std::declval<model::Proposal>().transactions)();
+  std::vector<shared_model::proto::Transaction> transactions;
   for (const auto &tx : request->transactions()) {
-    transactions.push_back(*factory_.deserialize(tx));
+    transactions.push_back(shared_model::proto::Transaction(tx));
```
Comment (nickaleks) [inline irohad/ordering/impl/ordering_gate_transport_grpc.cpp]: consider using emplace_back

File: irohad/simulator/impl/simulator.cpp [hunk 6]
```diff
@@ -38,9 +38,8 @@ namespace iroha {
           crypto_provider_(std::move(crypto_provider)) {
       log_ = logger::log("Simulator");
       ordering_gate->on_proposal().subscribe(
-          proposal_subscription_, [this](model::Proposal old_proposal) {
-            this->process_proposal(shared_model::proto::Proposal(
-                shared_model::proto::from_old(old_proposal)));
+          proposal_subscription_, [this](std::shared_ptr<shared_model::interface::Proposal> proposal) {
```
Comment (nickaleks) [inline irohad/simulator/impl/simulator.cpp]: Is something wrong with the formatting?

File: irohad/ordering/impl/ordering_gate_transport_grpc.cpp [hunk 7]
```diff
@@ -49,12 +57,16 @@ OrderingGateTransportGrpc::OrderingGateTransportGrpc(
       log_(logger::log("OrderingGate")) {}
 
 void OrderingGateTransportGrpc::propagateTransaction(
-    std::shared_ptr<const model::Transaction> transaction) {
+    const std::shared_ptr<const shared_model::interface::Transaction>
```
Comment (nickaleks) [inline irohad/ordering/impl/ordering_gate_transport_grpc.cpp]: Why pass pointer by reference?

File: irohad/ordering/impl/ordering_gate_transport_grpc.hpp [hunk 8]
```diff
@@ -39,15 +44,15 @@ namespace iroha {
                               ::google::protobuf::Empty *response) override;
 
       void propagateTransaction(
-          std::shared_ptr<const model::Transaction> transaction) override;
+          const std::shared_ptr<const shared_model::interface::Transaction>
```
Comment (nickaleks) [inline irohad/ordering/impl/ordering_gate_transport_grpc.hpp]: Same as above

**Pull request merged**